### PR TITLE
Fix nodejs for Armv6

### PR DESCRIPTION
**Pull request by Blakko**
First attempt to support the nodejs install on armv6 devices.

Please test it!

I took some of the lines from the node-red installation script.  I'm not sure what to do with regard to the license

Signed-off-by: Fabio Tacchelli fabio.tacchelli@gmail.com (github: Blakko)

---
This change is [Reviewable](https://reviewable.io/reviews/openhab/openhabian/277)

### CODE REVIEW

**Comment by ThomDietrich** [comment]
Hey @Blakko 👋  looks good on first sight. I'll do some testing in the next couple of days ;) THANKS!

File: functions/nodejs-apps.sh [hunk 1]
```diff
@@ -2,14 +2,24 @@
 
 nodejs_setup() {
   if ! command -v npm &>/dev/null; then
```
Comment (ThomDietrich) [inline functions/nodejs-apps.sh]: This is bad design (by me).
Please turn around:

```shell
if command -v npm &>/dev/null; then
  return 0
fi

FAILED=0
...
```

**Comment by ThomDietrich** [comment]
For reference: https://github.com/sdesalas/node-pi-zero

File: functions/nodejs-apps.sh [hunk 2]
```diff
@@ -2,14 +2,24 @@
...
-    echo -n "$(timestamp) [openHABian] Installing Node.js (prerequisite for other packages)... "
     FAILED=0
-    cond_redirect wget -O /tmp/nodejs-v7.x.sh https://deb.nodesource.com/setup_7.x || FAILED=1
-    cond_redirect bash /tmp/nodejs-v7.x.sh || FAILED=1
-    if [ $FAILED -eq 1 ]; then echo "FAILED (nodejs preparations)"; exit 1; fi
-    #cond_redirect apt update # part of the node script above
-    cond_redirect apt -y install nodejs
-    if [ $? -ne 0 ]; then echo "FAILED (nodejs installation)"; exit 1; fi
+    if is_pione || is_pizero || is_pizerow; then
```
Comment (ThomDietrich) [inline functions/nodejs-apps.sh]: I don't like this condition. Would you please add a is_armv6l function here: https://github.com/openhab/openhabian/blob/master/functions/helpers.sh#L70

File: functions/nodejs-apps.sh [hunk 3]
```diff
@@ -2,14 +2,24 @@
...
+      echo -n "$(timestamp) [openHABian] Install Node.js for Armv6 (prerequisite for other packages)... "
+      f=$(curl -sL https://nodejs.org/download/release/latest-boron/ | grep "armv6l.tar.gz" | cut -d '"' -f 2)
+      cond_redirect curl -sL -o node.tgz https://nodejs.org/download/release/latest-boron/$f 2>&1 || FAILED=1
+      if [ $FAILED -eq 1 ]; then echo "FAILED (nodejs preparations)"; exit 1; fi
+      # unpack it into the correct places
+      cond_redirect sudo tar -zxf node.tgz --strip-components=1 -C /usr 2>&1
```
Comment (ThomDietrich) [inline functions/nodejs-apps.sh]: remove sudo

File: functions/nodejs-apps.sh [hunk 4]
```diff
@@ -2,14 +2,24 @@
...
-    cond_redirect wget -O /tmp/nodejs-v7.x.sh https://deb.nodesource.com/setup_7.x || FAILED=1
-    cond_redirect bash /tmp/nodejs-v7.x.sh || FAILED=1
-    if [ $FAILED -eq 1 ]; then echo "FAILED (nodejs preparations)"; exit 1; fi
-    #cond_redirect apt update # part of the node script above
-    cond_redirect apt -y install nodejs
-    if [ $? -ne 0 ]; then echo "FAILED (nodejs installation)"; exit 1; fi
+    if is_pione || is_pizero || is_pizerow; then
+      echo -n "$(timestamp) [openHABian] Install Node.js for Armv6 (prerequisite for other packages)... "
+      f=$(curl -sL https://nodejs.org/download/release/latest-boron/ | grep "armv6l.tar.gz" | cut -d '"' -f 2)
+      cond_redirect curl -sL -o node.tgz https://nodejs.org/download/release/latest-boron/$f 2>&1 || FAILED=1
```
Comment (ThomDietrich) [inline functions/nodejs-apps.sh]: please use wget

**Comment by Blakko** [comment]
Note: missing usage of wget

File: functions/nodejs-apps.sh [hunk 5]
```diff
@@ -1,16 +1,18 @@
 #!/usr/bin/env bash
 
 nodejs_setup() {
-  if ! command -v npm &>/dev/null; then
+  if command -v npm &>/dev/null; then
+    return 0
+  else
```
Comment (ThomDietrich) [inline functions/nodejs-apps.sh]: the else block isn't needed. just end with `fi`. It's a small change but will make everything a bit clearer

Comment (ThomDietrich) [reply functions/nodejs-apps.sh]: Nevermind about this minor detail. I'll merge now

**Comment by ThomDietrich** [comment]
Thanks! Great addition with a very positive impact. It would be great to see more contributions by you! There are many things we could improve in openHABian! 👋

**Comment by ThomDietrich** [comment]
Correction: I'd still need you to sing-off your work. https://github.com/openhab/openhabian/blob/master/CONTRIBUTING.md

Add another commit with the sing-off or add it to the PR description above.

A sign off could look like:

```
Signed-off-by: Thomas Dietrich <thomas.dietrich@tu-ilmenau.de>
```

**Comment by Blakko** [comment]
I updated the PR description, let me know if it's ok!

**Pull request merged**